### DISCUSSION TITLE
Enable upload mask

### DISCUSSION THREAD

**Discussion by xVOLDYx** [lllyasviel/Fooocus#1960] [Q&A] [answered]
I downloaded the Fooocus_win64_2-1-791 few weeks ago and now I open it says version 2.1.862. I assume it was auto upadated and I have the latest version.

I do a lot of inpainting and was looking for speedup the masking by uploading the mask. But I couldn't find a way in default UI. Then I found this [https://github.com/xhoxye/Fooocus-up_mask](url) and this [https://github.com/lllyasviel/Fooocus/pull/1645](url)
I tried to replace webui.py and async_worker.py in my directory. I got the UI with "Inpaint advanced" tab but after that Fooocus did not generate any images.

Here is the CMD output

> G:\Program Files\Fooocus_win64_2-1-791>.\python_embeded\python.exe -s Fooocus\entry_with_update.py
> Already up-to-date
> Update succeeded.
> [System ARGV] ['Fooocus\\entry_with_update.py']
> Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
> Fooocus version: 2.1.862
> Running on local URL:  http://127.0.0.1:7865
> 
> To create a public link, set `share=True` in `launch()`.
> Total VRAM 12288 MB, total RAM 16180 MB
> Set vram state to: NORMAL_VRAM
> Always offload VRAM
> Device: cuda:0 NVIDIA GeForce RTX 3060 : native
> VAE dtype: torch.bfloat16
> Using pytorch cross attention
> Refiner unloaded.
> model_type EPS
> UNet ADM Dimension 2816
> Traceback (most recent call last):
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\gradio\routes.py", line 488, in run_predict
>     output = await app.get_blocks().process_api(
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\gradio\blocks.py", line 1431, in process_api
>     result = await self.call_function(
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\gradio\blocks.py", line 1103, in call_function
>     prediction = await anyio.to_thread.run_sync(
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\anyio\to_thread.py", line 33, in run_sync
>     return await get_asynclib().run_sync_in_worker_thread(
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
>     return await future
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
>     result = context.run(func, *args)
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\gradio\utils.py", line 707, in wrapper
>     response = f(*args, **kwargs)
>   File "G:\Program Files\Fooocus_win64_2-1-791\Fooocus\modules\advanced_parameters.py", line 23, in set_all_advanced_parameters
>     disable_preview, adm_scaler_positive, adm_scaler_negative, adm_scaler_end, adaptive_cfg, sampler_name, \
> ValueError: not enough values to unpack (expected 35, got 32)
> Using pytorch attention in VAE
> Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
> Using pytorch attention in VAE
> extra {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids', 'cond_stage_model.clip_l.text_projection', 'cond_stage_model.clip_l.logit_scale'}
> Base model loaded: G:\Program Files\Fooocus_win64_2-1-791\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors
> Request to load LoRAs [['sd_xl_offset_example-lora_1.0.safetensors', 0.1], ['None', 1.0], ['None', 1.0], ['None', 1.0], ['None', 1.0]] for model [G:\Program Files\Fooocus_win64_2-1-791\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors].
> Loaded LoRA [G:\Program Files\Fooocus_win64_2-1-791\Fooocus\models\loras\sd_xl_offset_example-lora_1.0.safetensors] for UNet [G:\Program Files\Fooocus_win64_2-1-791\Fooocus\models\checkpoints\juggernautXL_version6Rundiffusion.safetensors] with 788 keys at weight 0.1.
> Fooocus V2 Expansion: Vocab with 642 words.
> Fooocus Expansion engine loaded for cuda:0, use_fp16 = True.
> Requested to load SDXLClipModel
> Requested to load GPT2LMHeadModel
> Loading 2 new models
> [Fooocus Model Management] Moving model(s) has taken 0.42 seconds
> App started successful. Use the app with http://127.0.0.1:7865/ or 127.0.0.1:7865
> [Parameters] Adaptive CFG = None
> [Parameters] Sharpness = 2
> [Parameters] ADM Scale = None : None : None
> [Parameters] CFG = 4.0
> [Parameters] Seed = 7567763683669740930
> Traceback (most recent call last):
>   File "G:\Program Files\Fooocus_win64_2-1-791\Fooocus\modules\async_worker.py", line 836, in worker
>     handler(task)
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>   File "G:\Program Files\Fooocus_win64_2-1-791\python_embeded\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
>     return func(*args, **kwargs)
>   File "G:\Program Files\Fooocus_win64_2-1-791\Fooocus\modules\async_worker.py", line 353, in handler
>     if advanced_parameters.overwrite_step > 0:
> TypeError: '>' not supported between instances of 'NoneType' and 'int'
> Total time: 14.94 seconds 

How can I solve this? (I'm totally noob and couldn't find anywhere else to ask for help)

Thank you!

**Comment by psydox** [comment, marked as answer]
The upload mask is already available since "v2.1.860".  Check the screenshot of how to access it.

![Screenshot 2024-01-18 021247](https://github.com/lllyasviel/Fooocus/assets/5681500/282f92ee-f10e-4a74-a174-7f50e10f1829)

**Reply by xVOLDYx** [reply]
Thank you so much :)

**Reply by creamlabsAI** [reply]
Thank you as well!

**Comment by LiveFreeDead** [comment]
Any way to enable it by default... it's been a few months and the feature works great.

**Reply by mashb1t** [reply]
@LiveFreeDead you can change the visibility of the corresponding elements to true by default and check the checkboxes as true on startup
